### PR TITLE
Fix #319 Failing to parse some PackageJson Person fields

### LINKED ISSUE #319: Failing to parse some Person PackageJson fields
This results in an assertion during debug mode, and also a failure to properly display those fields in the UI

### PR DESCRIPTION
**Pull request by mousetraps**
Fix issues while parsing Person package.json fields
* we were not properly handling cases when the name, email, and url fields
      were out of order.
* fix assertion check when result is empty
* add relevant test cases

### CODE REVIEW

File: Nodejs/Tests/NpmTests/PackageJsonTests.cs [hunk 1]
```diff
@@ -108,6 +113,10 @@ public class PackageJsonTests : AbstractPackageJsonTests {
    ""name"": ""mypackage"",
    ""version"": ""0.7.0"",
    ""description"": ""Sample package for CommonJS. This package demonstrates the required elements of a CommonJS package."",
+   ""author"": {
```
Comment (paulvanbrenk) [inline Nodejs/Tests/NpmTests/PackageJsonTests.cs]: Add more test cases with only a single field and with an empty author element.

File: Nodejs/Product/Npm/SPI/Person.cs [hunk 2]
```diff
@@ -71,9 +83,9 @@ internal class Person : IPerson {
             // Verify we are parsing correctly
             try {
                 var jObject = JObject.Parse(source);
-                Debug.Assert(((string)jObject["name"] ?? string.Empty) == Name, string.Format("Failed to parse name from {0}", source));
-                Debug.Assert(((string)jObject["email"] ?? string.Empty) == Email, string.Format("Failed to parse email from {0}", source));
-                Debug.Assert(((string)jObject["url"] ?? string.Empty) == Url, string.Format("Failed to parse url from {0}", source));
+                Debug.Assert((string)jObject["name"] == Name, string.Format("Failed to parse name from {0}", source));
```
Comment (paulvanbrenk) [inline Nodejs/Product/Npm/SPI/Person.cs]: Why bother with this... we're extracting them using the regex, do we still care what the json parser makes of it...

Comment (mousetraps) [inline Nodejs/Product/Npm/SPI/Person.cs]: We don't care unless we're doing it wrong. It's nice to keep this codepath around in an #if DEBUG section and assert we're getting the same results as our own parser in case there are any inputs we're not handling properly (which is how we managed to track down this bug in the first place.)

**Comment by paulvanbrenk** [comment]
:+1:

File: Nodejs/Tests/NpmTests/PackageJsonTests.cs (hunk 1 above)
Comment (mousetraps) [inline Nodejs/Tests/NpmTests/PackageJsonTests.cs]: good call. added a few more.
